flask_app/models.py

```diff
@@ -88,7 +88,11 @@ def errored(self):
 class Session(db.Model, TypenameMixin, StatusPredicatesMixin, HasRelatedMixin, HasSubjectsMixin, UserDetailsMixin):
 
     id = db.Column(db.Integer, primary_key=True)
-    logical_id = db.Column(db.String(256), index=True)
+    logical_id = db.Column(db.String(256), unique=True, index=True)
+
+    parent_logical_id = db.Column(db.String(256), db.ForeignKey('session.logical_id'), default=None)
```

Comment:
@maormarcus we need to index this field... We are going to search children according to it